src/Admin/SiteHealth.php

```diff
@@ -92,6 +91,32 @@ private function get_badge_label() {
 		return esc_html__( 'AMP', 'amp' );
 	}
 
+	/**
+	 * Ajax handler for async site status check.
+	 *
+	 * @return void
+	 */
+	public function ajax_site_status() {
+
+		check_ajax_referer( 'health-check-site-status' );
+
+		if ( ! current_user_can( 'view_site_health_checks' ) ) {
+			wp_send_json_error();
+		}
+
+		$feature  = ( ! empty( $_POST['feature'] ) ) ? sanitize_key( $_POST['feature'] ) : '';
+		$function = sprintf(
+			'json_test_%s',
+			$feature
+		);
+
+		if ( ! method_exists( $this, $function ) || ! is_callable( [ $this, $function ] ) ) {
+			return;
+		}
+
+		call_user_func( [ $this, $function ] );
```

Comment:
Continuing from above, this could be simplified into something like (which would include removing the `json_test_amp_page_cache` method:

```suggestion
	 * Ajax handler for page cache status check.
	 *
	 * @return void
	 */
	public function ajax_site_status() {

		check_ajax_referer( 'health-check-site-status' );

		if ( ! current_user_can( 'view_site_health_checks' ) ) {
			wp_send_json_error();
		}

		wp_send_json_success( $this->page_cache() );
```